soc/raspberrypi/rpi_pico/common/pinctrl_soc.h

```diff
@@ -7,7 +7,7 @@
 #ifndef ZEPHYR_SOC_ARM_RPI_PICO_RP2_PINCTRL_SOC_H_
 #define ZEPHYR_SOC_ARM_RPI_PICO_RP2_PINCTRL_SOC_H_
 
-#include <zephyr/dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/rpi-pico-pinctrl-common.h>
```

Comment:
> An incomplete header is being loaded.

Correct! Also.... it's not an issue. The necessary symbols are defined, demonstrably this works as intended.

> It would be better to...

This is a subjective opinion presented as fact. I disagree with the proposal. `soc/<VENDOR>/common/pinctrl_soc.h` is an idiom used by multiple SoC vendors in the codebase.

Reply:
> > An incomplete header is being loaded.
> 
> Correct! Also.... it's not an issue. The necessary symbols are defined, demonstrably this works as intended.
> 
> > It would be better to...
> 
> This is a subjective opinion presented as fact. I disagree with the proposal. `soc/<VENDOR>/common/pinctrl_soc.h` is an idiom used by multiple SoC vendors in the codebase.

Even how gently I express it, I must say this is an unmanifested problem.
If the issue is not fixed here, we'll be pushing the work of fixing it onto someone else.
I can't approve it if this ISSUE is left.

> This is a subjective opinion presented as fact. I disagree with the proposal. `soc/<VENDOR>/common/pinctrl_soc.h` is an idiom used by multiple SoC vendors in the codebase.

How to fix it is a matter of preference, so I'll leave you.
(Of course, whether it is approvable or not, it is a separate problem.)
As an additional comment, creating `common` under `soc` (and `boards`) is not usual.
In most cases, only the soc series directory is created.